### DISCUSSION TITLE
Project HÆLTH Blog #7

### DISCUSSION THREAD

**Discussion by OmarMoujahed** [ldcdorn/haelth#64] [General]
# Project HÆLTH 🌱
In our blog we present our quality tree with 5 quality features and our Architecture Decision Records (ADR)

## Our quality features
![image](https://github.com/user-attachments/assets/bc802167-3297-4fdf-957e-210139f32ea0)

### Quality tree as a guide
The quality of the system will be improved based on the defined characteristics and scenarios. Our focus is on the prioritized characteristics: performance, ease of use, reliability, safety and maintainability.

### 1.) Fitness (Mental/Physically):
Fitness, both physical and mental, is defined by scenarios A10, A20 and A30.

### 2.) Scalebility:
The system’s ability to scale with growing number of users and data is supported by scenarios B10, B20 and B30.

### 3.) Reliability:
####System reliability is supported by robust architecture and data integrity as described in C10 and C20 scenarios.

## Strategies and priorization
In every project, it is essential to use resources effectively. Therefore, it is advisable to focus on one or two scenarios with high priority.

#### Safety:
Encrypt data at the client before transmission and decrypt only at the recipient.

#### Performance:
Implementation of validation mechanisms to prevent erroneous data.

### Architecture Decision Records
The HÆLTH application stores user data, such as previous workouts, meal plans, and meditation sessions. Data can be stored in various ways, but we debated between using a local database or saving it in a CSV file. We decided to use a local database because it is faster and more efficient than CSV files, especially when handling larger amounts of data.

## Refactoring parts of our code
Apart from our progress and improvements mentioned above, we also decided to refactor and rewrite some of our code. Reason for that is the different ways you can create an UI in Android Studio (eg. Cards or Compose). Since we did not define what to use for UI, different parts of it are created in different ways. Keeping this throughout the project would cause a lot of frustration and time wasted, so we decided to update and refactor the "outdated" parts of our app, to have an easier, more structured and long-lasting work experience in the future.

**Comment by ktoffel999** [comment]
Der Qualitätsbaum ist gut strukturiert und adressiert wichtige Systemmerkmale wie Leistung, Benutzerfreundlichkeit, Zuverlässigkeit, Sicherheit und Wartbarkeit. Die Fokussierung auf bestimmte Szenarien und die Strategieentscheidungen sind zielführend. Die Entscheidung zur Refaktorisierung für eine einheitliche Codebasis ist ebenso positiv. Insgesamt zeigt der Ansatz ein klares Verständnis für Qualität und Zukunftsfähigkeit des Systems.
MfG das Jump and Run Team

**Comment by themsgcentre** [comment]
Wir finden eure Ausarbeitung der Szenarien und euren Qualitätsbaum sehr gut strukturiert und detailliert. Die Ausführlichkeit und Klarheit, mit der das Thema beleuchtet wird, ist bemerkenswert. Besonders gefällt uns, wie praxisnahe Tipps und fundierte Informationen kombiniert wurden. Ein großartiger Fortschritt!

Viele Grüße, 
Projekt TDC (HM)
